### PR TITLE
Deprecate gRPC-Plugin support

### PR DESCRIPTION
**Pull request by h4shk4t**
## Which problem is this PR solving?
- Part of [#4647 ](https://github.com/jaegertracing/jaeger/issues/4647)

## Description of the changes
- Remove sidecar plugin support based on hashicorp go-plugin library.

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: plugin/storage/grpc/config/config.go [hunk 1]
```diff
@@ -15,42 +15,31 @@
...
-	remoteConn            *grpc.ClientConn
-}
+// type Configuration struct {
+// 	PluginBinary            string `yaml:"binary" mapstructure:"binary"`
+// 	PluginConfigurationFile string `yaml:"configuration-file" mapstructure:"configuration_file"`
+// 	PluginLogLevel          string `yaml:"log-level" mapstructure:"log_level"`
+// 	RemoteServerAddr        string `yaml:"server" mapstructure:"server"`
+// 	RemoteTLS               tlscfg.Options
+// 	RemoteConnectTimeout    time.Duration `yaml:"connection-timeout" mapstructure:"connection-timeout"`
+// 	TenancyOpts             tenancy.Options
```
Comment (yurishkuro) [inline plugin/storage/grpc/config/config.go:36]: These are not plugin options, they are for remote grpc storage that we will continue to support

**Review by yurishkuro — COMMENTED** [review]
- please delete code, not comment out
- changes look like in the right direction, but they don't compile

File: plugin/storage/grpc/config/config.go [hunk 2]
```diff
@@ -34,21 +31,16 @@ import (
...
-	RemoteConnectTimeout    time.Duration `yaml:"connection-timeout" mapstructure:"connection-timeout"`
-	TenancyOpts             tenancy.Options
+	PluginLogLevel       string `yaml:"log-level" mapstructure:"log_level"`
+	RemoteServerAddr     string `yaml:"server" mapstructure:"server"`
+	RemoteTLS            tlscfg.Options
+	RemoteConnectTimeout time.Duration `yaml:"connection-timeout" mapstructure:"connection-timeout"`
+	TenancyOpts          tenancy.Options
 
 	pluginHealthCheck     *time.Ticker
 	pluginHealthCheckDone chan bool
```
Comment (yurishkuro) [inline plugin/storage/grpc/config/config.go:43]: I assume these should also be gone?

File: plugin/storage/grpc/options.go [hunk 3]
```diff
@@ -28,16 +27,12 @@ import (
 )
 
 const (
-	pluginBinary             = "grpc-storage-plugin.binary"
-	pluginConfigurationFile  = "grpc-storage-plugin.configuration-file"
 	pluginLogLevel           = "grpc-storage-plugin.log-level"
```
Comment (yurishkuro) [inline plugin/storage/grpc/options.go:30]: this looks like it's related to sidecar plugin model, not to remote grpc-storage

File: plugin/storage/grpc/shared/interface.go [hunk 4]
```diff
@@ -15,26 +15,13 @@
 package shared
```
Comment (yurishkuro) [inline plugin/storage/grpc/shared/interface.go:15]: Do we need any of these interfaces?

**Review by yurishkuro — APPROVED** [review]

File: plugin/storage/grpc/grpc.go [hunk 5]
```diff
@@ -1,46 +0,0 @@
...
-)
-
-// Serve creates a plugin configuration using the implementation of StoragePlugin and then serves it.
-func Serve(services *shared.PluginServices) {
-	ServeWithGRPCServer(services, plugin.DefaultGRPCServer)
-}
-
-// ServeWithGRPCServer creates a plugin configuration using the implementation of StoragePlugin and
-// function to create grpcServer, and then serves it.
-func ServeWithGRPCServer(services *shared.PluginServices, grpcServer func([]grpc.ServerOption) *grpc.Server,
```
Comment (SOF3) [inline plugin/storage/grpc/grpc.go:31]: Is this removing the sidecar model or removing grpc plugin in general? What is the new correct way to start a grpc plugin if the StorageGRPCPlugin is also removed?

**Pull request merged**